src/app/modules/item/components/item-children-edit/item-children-edit.component.ts

```diff
@@ -107,10 +112,19 @@ export class ItemChildrenEditComponent implements OnChanges {
 
   resetScoreWeight(): void {
     this.data = this.data.map(c => ({ ...c, scoreWeight: DEFAULT_SCORE_WEIGHT }));
+    this.onChildrenListUpdate();
     this.onScoreWeightChange();
   }
 
   onScoreWeightChange(): void {
     this.childrenChanges.emit(this.data);
   }
+
+  onChildrenListUpdate(): void {
+    this.addedItemIds = this.getAddedItemIds();
+  }
+
+  getAddedItemIds(): string[] {
+    return this.data.map(item => item.id).filter(isNotUndefined);
```

Comment:
If used at a single place, I prefer you put this code directly it is used. (this should not have used outside setting the new value to this.addedItemIds).
(alternatively, you call your function "recomputeAddedItemIds" and this function assign the value directly.

Reply:
Fixed